finat/tensor_product.py

```diff
@@ -179,12 +174,11 @@ def dual_basis(self):
         # Naming as _merge_evaluations above
         alphas = [factor.get_indices() for factor in self.factors]
         zetas = [factor.get_value_indices() for factor in self.factors]
-        # Index the factors by so that we can reshape into index-shape
-        # followed by value-shape
-        qis = [q[alpha + zeta] for q, alpha, zeta in zip(qs, alphas, zetas)]
         Q = gem.ComponentTensor(
-            reduce(gem.Product, qis),
-            tuple(chain(*(alphas + zetas)))
+            # Index the factors by so that we can reshape into index-shape
```

Comment:
not sure this comment makes sense - index the factors by what?

Reply:
```suggestion
            # Index the factors by basis function and value indices so that we can reshape into index-shape
```